extmod/modtls_mbedtls.c

```diff
@@ -91,24 +92,6 @@ STATIC mp_obj_t ssl_socket_make_new(mp_obj_ssl_context_t *ssl_context, mp_obj_t
 /******************************************************************************/
 // Helper functions.
 
-STATIC mp_obj_t read_file(mp_obj_t self_in) {
```

Comment:
@felixdoerre why would you delete this? I thought this was better (i.e. less code size/memory) than reading the files in `ssl.py` module? Have you compared both implementations?

Reply:
I didn't until now, because for me the comparison was "have it implement in c" vs "don't have it in c", as the goal is to only have the tls module and eventually make `ssl.py` optional.

However I've just now tested it and it seems that the implementation in python is indeed smaller (and handler an error in `f.read` correctly, by closing the file:
```
   text    data     bss     dec     hex filename
 718248   75912    7056  801216   c39c0 build-standard/micropython (as implemented in this PR)
 718568   75944    7056  801568   c3b20 build-standard/micropython (with the file read logic moved back to C)
```

Reply:
Then I guess it makes sense to move it to `ssl.py`, thanks for testing it 👍🏼

Reply:
Good to know that it's smaller in Python!